### PR TITLE
 use datafusion plan_to_sql function instead of expr_to_sql

### PR DESCRIPTION
**Pull request by hozan23**
Hello,

This PR addresses the the issue #103

### CODE REVIEW

**Comment by hozan23** [comment]
We need to use datafusion version 42  to make this PR work because of this patch https://github.com/apache/datafusion/commit/ddbdf4bf7f215c01aca435f8edd00412e84aab60. 

We will wait until we merge the spiceai branch that has already updated to Datafusion version 42 #112

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**
